tests/unit/OrgPerson.spec.ts

```diff
@@ -542,114 +567,591 @@ describe('Org/Person component for a Correction filing', () => {
+    expect(wrapper.find('.person-name label').text()).toBe('Person\'s Name')
+    expect(wrapper.find('.confirm-name-change').exists()).toBe(false)
+    expect(wrapper.find('.edit-business-number').exists()).toBe(false)
+    expect(wrapper.find('.show-business-number').exists()).toBe(false)
+    expect(wrapper.find('.org-look-up').exists()).toBe(false)
+    expect(wrapper.find('.org-manual-entry').exists()).toBe(false)
+    expect(wrapper.find('.other-edit-org').exists()).toBe(false)
+    expect(wrapper.find('.incorporation-number').exists()).toBe(false)
+    expect(wrapper.find('.email-address .sub-header').text()).toBe('Email Address')
+    expect(wrapper.find('#proprietor-partner-email').element['value']).toBe('partner@example.com')
+    expect(wrapper.find('.roles').exists()).toBe(false)
+    expect(wrapper.find('.mailing-address .sub-header').text()).toBe('Mailing Address')
+    expect(wrapper.find('.delivery-address .sub-header').text()).toBe('Delivery Address')
+    expect(wrapper.find(removeButtonSelector).exists()).toBe(true)
+
+    wrapper.destroy()
+  })
+
+  it('Changes existing GP partner - organization', () => {
+    store.state.stateModel.tombstone.entityType = 'GP'
+    store.state.resourceModel = ChangePartnershipResource
+
+    const wrapper = mount(OrgPerson, {
+      store,
+      vuetify,
+      propsData: {
+        currentOrgPerson: PartnerOrgData,
         activeIndex: 0,
         currentCompletingParty: null
       }
     })
 
-    expect(wrapper.findAll('.sub-header').at(0).text()).toBe('Business or Corporation Name')
+    expect(wrapper.find('.add-person-header').exists()).toBe(false)
+    expect(wrapper.find('.add-org-header').text()).toBe('Edit Business or Corporation')
+    expect(wrapper.find('.person-name label').exists()).toBe(false)
+    expect(wrapper.find('.confirm-name-change').exists()).toBe(false)
+    expect(wrapper.find('.edit-business-number .sub-header').text()).toBe('Business Number')
+    expect(wrapper.find('.show-business-number').exists()).toBe(false)
+    expect(wrapper.find('.org-look-up').exists()).toBe(false)
+    expect(wrapper.find('.org-manual-entry').exists()).toBe(false)
+    expect(wrapper.find('.other-edit-org .sub-header').text()).toBe('Business or Corporation Name')
+    expect(wrapper.find('.incorporation-number .sub-header').text()).toBe('Incorporation/Registration Number:')
+    expect(wrapper.find('.incorporation-number .sub-header-text').text()).toBe('FM1234567')
+    expect(wrapper.find('.email-address .sub-header').text()).toBe('Email Address')
+    expect(wrapper.find('#proprietor-partner-email').element['value']).toBe('partner@example.com')
+    expect(wrapper.find('.roles').exists()).toBe(false)
+    expect(wrapper.find('.mailing-address .sub-header').text()).toBe('Mailing Address')
+    expect(wrapper.find('.delivery-address .sub-header').text()).toBe('Delivery Address')
+    expect(wrapper.find(removeButtonSelector).exists()).toBe(true)
```

Comment:
Reducing duplication via for loop, etc is left to the future.

Reply:
Yes, exactly.

If nothing else comes up today then I will look at parameterizing these tests to clean them up.

Reply:
I noticed you use class and components in verifying with expect is there a benefit to using one over the other?

Reply:
(Line 1154 isn't a component; it's a string constant in this test suite.)

As for Vuetify components, since they are "exploded" at render-time, sometimes the ID is assigned to the internal input element, so I can't select some of the things I want. In that case, the classname is usually assigned to the outer div and then I can use that and the internal Vuetify classes (eg, __input) to select things. This is also the case for selecting a thing to click or set a value on.

As for, more generally, why assign classnames instead of IDs? Because IDs should be unique on a page. For testing, classes usually work out better. An alternative that we don't use often is `data-test="xxx"`, though I've seen this best used in loops to select rendered data. It's all a bit of a gray area :)

Reply:
Okay that makes sense. By using class names you make it more general and thus a less fragile test.

Reply:
That's not it. By using classnames I reduce the change of an ID conflict now or down the line (though I search globally to try to prevent initial conflicts).

Actually, we want fragile unit tests -- they're a major safety check when we implement new/changed things. Sometimes the tests have to be fixed, but other times they point to something we broke :)

Reply:
OrgPerson test results:
```
> business-edit-ui@3.3.2 test:unit
> vue-cli-service test:unit --testPathPattern "OrgPerson"

PASS tests/unit/OrgPerson.spec.ts (5.318s)
  Org/Person component for a BEN Correction filing
    ✓ Loads the component and sets data for person (232ms)
    ✓ Loads the component and sets data for org (88ms)
    ✓ Displays edit form for person (161ms)
    ✓ Displays add form for org (92ms)
    ✓ Enables Remove button in edit mode (87ms)
    ✓ Disables Remove button in create mode (78ms)
    ✓ Emits "remove" event when clicking Remove button (76ms)
    ✓ Emits "reset" event when clicking Done button and org has not changed (113ms)
    ✓ Emits "addEdit" event when clicking Done button and org has changed (114ms)
    ✓ Emits "reset" event when clicking Cancel button (105ms)
    ✓ Does not display error message when user enters valid org name (77ms)
    ✓ Displays error message when user enters invalid org name (75ms)
    ✓ Does not display error message when user enters valid person names (113ms)
    ✓ Displays error message when user does not enter person names (175ms)
    ✓ Displays error message when user enters person names that are too long (129ms)
    ✓ Shows popup if there is an existing completing party (121ms)
    ✓ Emits "removeCpRole" and "addEdit" events on reassigning the Completing Party (161ms)
    ✓ Displays errors and does not submit form when clicking Done button and form is invalid (99ms)
  Org/Person component for SP/GP filings
    ✓ 1. changeOfRegistration - adds new GP partner-person (100ms)
    ✓ 2. changeOfRegistration - adds new GP partner-organization (108ms)
    ✓ 3. changeOfRegistration - changes existing SP proprietor-person (97ms)
    ✓ 4. changeOfRegistration - changes existing SP proprietor-organization (97ms)
    ✓ 5. changeOfRegistration - changes existing GP partner-person (96ms)
    ✓ 6. changeOfRegistration - changes existing GP partner - organization (88ms)
    ✓ 7. conversion - adds new SP proprietor-person (107ms)
    ✓ 8. conversion - adds new SP proprietor-organization (108ms)
    ✓ 9. conversion - adds new GP partner-person (99ms)
    ✓ 10. conversion - adds new GP partner-organization (103ms)
    ✓ 11. conversion - changes existing SP proprietor - person (104ms)
    ✓ 12. conversion - changes existing SP proprietor-organization (99ms)
    ✓ 13. conversion - changes existing GP partner-person (91ms)
    ✓ 14. conversion - changes existing GP partner-organization (86ms)

Test Suites: 1 passed, 1 total
Tests:       32 passed, 32 total
Snapshots:   0 total
Time:        5.851s, estimated 13s
Ran all test suites matching /OrgPerson/i.
```